tests/test-unattended-config.c

```diff
@@ -0,0 +1,164 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*- */
+/*
+ * Copyright © 2018 Endless Mobile, Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
+ * 02111-1307, USA.
+ */
+#include "config.h"
+#include <locale.h>
+#include <string.h>
+#include <glib/gstdio.h>
+
+#include "gis-unattended-config.h"
+
+typedef struct {
+    gchar *tmpdir;
+} Fixture;
+
+static void
+fixture_set_up (Fixture *fixture,
+                gconstpointer user_data)
+{
+  g_autoptr(GError) error = NULL;
+
+  fixture->tmpdir = g_dir_make_tmp ("eos-installer.XXXXXX", &error);
+  g_assert_no_error (error);
+  g_assert_nonnull (fixture->tmpdir);
+}
+
+static void
+rm_r (const gchar *path)
+{
+  g_autoptr(GFile) file = g_file_new_for_path (path);
+  g_autoptr(GFileEnumerator) enumerator = NULL;
+  GFile *child = NULL;
+  GError *error = NULL;
+
+  enumerator = g_file_enumerate_children (
+      file, NULL, G_FILE_QUERY_INFO_NOFOLLOW_SYMLINKS, NULL, &error);
+  g_assert_no_error (error);
+
+  while (g_file_enumerator_iterate (enumerator, NULL, &child, NULL, &error) &&
+         child != NULL)
+    {
+      if (!g_file_delete (child, NULL, &error))
+        {
+          g_autofree gchar *child_path = g_file_get_path (child);
+          g_warning ("Failed to delete %s: %s", child_path, error->message);
+          g_clear_error (&error);
+        }
+    }
+
+  if (error != NULL)
+    {
+      g_warning ("Error while enumerating %s: %s", path, error->message);
+      g_clear_error (&error);
+    }
+
+  if (!g_file_delete (file, NULL, &error))
+    {
+      g_warning ("Failed to delete %s: %s", path, error->message);
+      g_clear_error (&error);
+    }
+}
+
+static void
+fixture_tear_down (Fixture *fixture,
+                   gconstpointer user_data)
+{
+  rm_r (fixture->tmpdir);
+  g_clear_pointer (&fixture->tmpdir, g_free);
+}
+
+static void
+test_parse_empty (void)
+{
+  g_autofree gchar *empty_ini =
+    g_test_build_filename (G_TEST_DIST, "unattended/empty.ini", NULL);
```

Comment:
Nitpick: Instead of the faff of shipping an empty INI file, I’d probably just touch an empty file in the fixture tmpdir for this test. Choose your poison.